health/health_virtual.c

```diff
@@ -0,0 +1,168 @@
+              rrdhost_hostname(host), rrdcalc_chart_name(rcv), rrdcalc_name(rcv), rcv->value
+              );
+
+        buffer_json_member_add_double(wb, "db_lookup", rcv->value);
+    }
+
+    if (unlikely(rcv->calculation)) {
+        rcv->calculation->value_at = at;
+
+        if (unlikely(!expression_evaluate(rcv->calculation))) {
+            // calculation failed
+            rcv->value = NAN;
+            rcv->run_flags |= RRDCALC_FLAG_CALC_ERROR;
+
+            netdata_log_debug(D_HEALTH, "Health (virtual) on host '%s', alarm '%s.%s': expression '%s' failed: %s",
+                  rrdhost_hostname(host), rrdcalc_chart_name(rcv), rrdcalc_name(rcv),
+                  rcv->calculation->parsed_as, buffer_tostring(rcv->calculation->error_msg)
+                  );
+            buffer_json_member_add_string(wb, "calc_error", buffer_tostring(rcv->calculation->error_msg));
+        } else {
+            rcv->run_flags &= ~RRDCALC_FLAG_CALC_ERROR;
+
+            netdata_log_debug(D_HEALTH, "Health (virtual) on host '%s', alarm '%s.%s': expression '%s' gave value "
+                  NETDATA_DOUBLE_FORMAT
+                  ": %s (source: %s)", rrdhost_hostname(host), rrdcalc_chart_name(rcv), rrdcalc_name(rcv),
+                  rcv->calculation->parsed_as, rcv->calculation->result,
+                  buffer_tostring(rcv->calculation->error_msg), rrdcalc_source(rcv)
+                  );
+
+            rcv->value = rcv->calculation->result;
+            buffer_json_member_add_double(wb, "calc", rcv->value);
+        }
+    }
+
+    if (likely(rcv->warning)) {
+        rcv->warning->value_at = at;
+
+        if (unlikely(!expression_evaluate(rcv->warning))) {
+            // calculation failed
+            rcv->run_flags |= RRDCALC_FLAG_WARN_ERROR;
+        } else {
+            rcv->run_flags &= ~RRDCALC_FLAG_WARN_ERROR;
+            buffer_json_member_add_double(wb, "warn", rcv->warning->result);
+            if (rcv->warning->result) {
+                rcv->status = RRDCALC_STATUS_WARNING;
+                vraised_warn = true;
+            }
+        }
+    }
+
+    if (likely(rcv->critical)) {
+        rcv->critical->value_at = at;
+
+        if (unlikely(!expression_evaluate(rcv->critical))) {
+            // calculation failed
+            rcv->run_flags |= RRDCALC_FLAG_CRIT_ERROR;
+        } else {
+            rcv->run_flags &= ~RRDCALC_FLAG_CRIT_ERROR;
+            buffer_json_member_add_double(wb, "crit", rcv->critical->result);
+            if (rcv->critical->result) {
+                rcv->status = RRDCALC_STATUS_CRITICAL;
+                vraised_crit = true;
+            }
+        }
+    }
+
+    if (!vraised_warn && !vraised_crit)
+        rcv->status = RRDCALC_STATUS_CLEAR;
+
+    buffer_json_object_close(wb);
+}
+
+void health_virtual(RRDHOST *host, BUFFER *wb, struct health_virtual *hv, int min_run_every) {
+
+    buffer_json_member_add_object(wb, "configuration"                  );
```

Comment:
We also have a space after `configuration` that is not necessary, unless we are going to add more data.